go/internal/services/keys/service.go

```diff
@@ -1,33 +1,48 @@
 package keys
 
 import (
+	"github.com/unkeyed/unkey/go/internal/services/ratelimit"
+	"github.com/unkeyed/unkey/go/internal/services/usagelimiter"
 	"github.com/unkeyed/unkey/go/pkg/cache"
-	"github.com/unkeyed/unkey/go/pkg/clock"
+	"github.com/unkeyed/unkey/go/pkg/clickhouse"
 	"github.com/unkeyed/unkey/go/pkg/db"
 	"github.com/unkeyed/unkey/go/pkg/otel/logging"
+	"github.com/unkeyed/unkey/go/pkg/rbac"
 )
 
+// Config holds the configuration for creating a new keys service instance.
 type Config struct {
-	Logger         logging.Logger
-	DB             db.Database
-	Clock          clock.Clock
-	KeyCache       cache.Cache[string, db.Key]
-	WorkspaceCache cache.Cache[string, db.Workspace]
+	Logger       logging.Logger        // Logger for service operations
+	DB           db.Database           // Database connection
+	RateLimiter  ratelimit.Service     // Rate limiting service
+	UsageLimiter usagelimiter.Service  // Usage limiting service
+	RBAC         *rbac.RBAC            // Role-based access control
+	Clickhouse   clickhouse.ClickHouse // Clickhouse for telemetry
+
+	KeyCache cache.Cache[string, db.FindKeyForVerificationRow] // Cache for key lookups
 }
 
 type service struct {
-	logger logging.Logger
-	db     db.Database
+	logger       logging.Logger
+	db           db.Database
+	raterLimiter ratelimit.Service
+	usageLimiter usagelimiter.Service
+	rbac         *rbac.RBAC
+	clickhouse   clickhouse.ClickHouse
+
 	// hash -> key
-	keyCache       cache.Cache[string, db.Key]
-	workspaceCache cache.Cache[string, db.Workspace]
```

Comment:
the reason for a separate workspace cache was that we might enforce quotas or introduce workspace level ratelimits when people exceed their quotas. Now this would be more difficult cause we'd need to cache it on each root key, right?

workspaces rarely changed, so we can set high stale timeouts and achieve goot cache hit rates

Reply:
yeah, thats fair I will add that cache back since its a quick lookup anyways